pkg/plugins/resources/yaml/main.go

```diff
@@ -52,28 +51,50 @@ func New(spec interface{}) (*Yaml, error) {
 		contentRetriever: &text.Text{},
 	}
 
-	newResource.spec.File = strings.TrimPrefix(newResource.spec.File, "file://")
-
-	err = newResource.Validate()
+	err = newResource.spec.Validate()
 	if err != nil {
 		return nil, err
 	}
 
+	newResource.files = make(map[string]string)
+	// File as unique element of newResource.files
+	if len(newResource.spec.File) > 0 {
+		newResource.files[strings.TrimPrefix(newResource.spec.File, "file://")] = ""
+	}
+	// Files
+	for _, filePath := range newResource.spec.Files {
+		newResource.files[strings.TrimPrefix(filePath, "file://")] = ""
+	}
+
 	return newResource, nil
+}
+
+func hasDuplicates(values []string) bool {
+	uniqueValues := make(map[string]string)
+	for _, v := range values {
+		uniqueValues[v] = ""
+	}
 
+	return len(values) != len(uniqueValues)
 }
 
 // Validate validates the object and returns an error (with all the failed validation messages) if it is not valid
-func (y *Yaml) Validate() error {
+func (s *Spec) Validate() error {
```

Comment:
👍 good thing to restrict the `Validate()` method to the scope it really validates